### PR TITLE
LoKI: Implement base class for panel

### PR DESCRIPTION
**Pull request by umithardal**
This PR introduces a base class for LoKI panel. This introduction removes the code repetitions in four files. The implemented class can be extended in the future to hold relevant functionalities.

### CODE REVIEW

File: nicos_ess/loki/gui/sampleconf.py [hunk 1]
```diff
@@ -219,7 +220,7 @@ def on_readDevsBtn_clicked(self):
             self._addRow('sam_y', self._readDev('sam_y'))
 
 
-class LokiSamplePanel(Panel):
+class LokiSamplePanel(Panel, ConnectionListener):
```
Comment (kmurica) [inline nicos_ess/loki/gui/sampleconf.py:223]: ConnectionListener already inherits from Panel. Might be an idea to rename the ConnectionListener to something else (a (bad) example: PanelWithConnectionListener) and just remove Panel in the extension here. Either that or just have ConnectionListener as a class attribute. This comment is also valid for LokiExperimentPanel.

Comment (kmurica) [reply nicos_ess/loki/gui/sampleconf.py:223]: LGTM to me otherwise.

Comment (umithardal) [reply nicos_ess/loki/gui/sampleconf.py:223]: I am thinking.

**Comment by umithardal** [comment]
@kmurica I believe I have fixed overall inheritance. This changes make LoKI panel easy to extend. Please confirm everything is fine.

File: nicos_ess/loki/gui/loki_panel.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+#   AÜC Hardal <umit.hardal@ess.eu>
+#
+# *****************************************************************************
+
+from nicos.clients.gui.panels import Panel
+
+
+class LokiPanelBase(Panel):
+    """
+    This is the base class for LoKI panel.
```
Comment (mattclarke) [inline nicos_ess/loki/gui/loki_panel.py:30]: The class is well named enough not to need a comment

**Comment by mattclarke** [comment]
This is good work - thanks!
Once I have tried it, I'll merge it

**Review by kmurica — APPROVED** [review]
Looks good to me but I agree with Matt that the docstring is unnecessary in this case.
So just remove it and if it is otherwise fine by Matt we can close this PR and merge it.

**Review by ebadkamil — APPROVED** [review]

**Pull request merged**